pkg/lang/ir/supervisor.go

```diff
@@ -60,22 +74,29 @@ func (g Graph) compileEntrypoint(root llb.State) llb.State {
 		return root
 	}
 	cmd := fmt.Sprintf("/var/envd/bin/envd-sshd --port %d --shell %s", config.SSHPortInContainer, g.Shell)
-	entrypoint := g.addNewProcess(root, "sshd", cmd)
+	entrypoint := g.addNewProcess(root, "sshd", cmd, nil)
+	var deps []string
+	if g.RuntimeInitScript != nil {
+		for i, command := range g.RuntimeInitScript {
+			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("init_%d", i), fmt.Sprintf("/bin/bash -c 'set -euo pipefail\n%s'", strings.Join(command, "\n")), nil)
+			deps = append(deps, fmt.Sprintf("init_%d", i))
+		}
+	}
 
 	if g.RuntimeDaemon != nil {
 		for i, command := range g.RuntimeDaemon {
-			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("daemon_%d", i), fmt.Sprintf("%s &\n", strings.Join(command, " ")))
+			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("daemon_%d", i), fmt.Sprintf("%s &\n", strings.Join(command, " ")), deps)
```

Comment:
I think we should emphasize their dependency relationship in the document later.

Reply:
Let's open an issue in envd-docs to keep track.

Reply:
https://github.com/tensorchord/envd-docs/issues/170